### PR TITLE
refactor(mission_planner): node refactoring

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description

- rename files
- use `/localization/kinematic_state` for the ego pose source
- small refactoring

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by takayuki5168** [comment]
@mitsudome-r @TakaHoribe Could you review this.
The reason to add `check_points_`, which is a member variable, to `plan_route`, which is a member function, is that we want to call `plan_route` function externally with mission_planner as a library.
image

**Review by TakaHoribe — APPROVED** [review]
It works well
image

**Comment by takayuki5168** [comment]
Thank you for your validation with psim.

I'll wait for @mitsudome-r 's approval and merge.

**Comment by takayuki5168** [comment]
@mitsudome-r kindly ping. Just approval is fine.

@TakaHoribe I think it's better that you or I or both become a code ower of the mission_planner package. What do you think?

File: planning/mission_planner/src/mission_planner_interface.cpp
Comment (mitsudome-r) [inline planning/mission_planner/src/mission_planner_interface.cpp:86]: Should there be transformation to `map_frame_` after this validation just in case?
Although it would be a rare case, since we are making map_frame_ configurable, we never know in which frame user would like to calculate route.

Something similar to goal_pose:
```
  const auto opt_transformed_ego_pose = transform_pose(ego_pose_, map_frame_);
  if (!opt_transformed_ego_pose) {
    RCLCPP_ERROR(get_logger(), "Failed to get vehicle pose in map frame. Aborting mission planning");
    return;
  }
  const auto transformed_ego_pose = opt_transformed_ego_pose.get();
```

**Review by mitsudome-r — COMMENTED** [review]
Thanks for the PR.
I had one small question regarding the following change. Please check my review comment.
> use /localization/kinematic_state for the ego pose source

File: planning/mission_planner/src/mission_planner_interface.cpp
Comment (takayuki5168) [reply planning/mission_planner/src/mission_planner_interface.cpp:86]: Ah, right.
I guess we have to apply the same transformation to ego pose in other planning modules. What do you think?

Comment (mitsudome-r) [reply planning/mission_planner/src/mission_planner_interface.cpp:86]: Yes, I think that would be a good idea. If it makes sense to do that as separate PR, I can approve this PR as is for now.

Comment (takayuki5168) [reply planning/mission_planner/src/mission_planner_interface.cpp:86]: Then P/C team and I will work on it with another PR. Could you approve please?

**Review by mitsudome-r — APPROVED** [review]

**Pull request merged**
